chargebee/src/main/java/com/chargebee/android/billingservice/BillingClientManager.kt

```diff
@@ -253,6 +255,102 @@ class BillingClientManager(context: Context) : PurchasesUpdatedListener {
+
+        billingClient?.queryProductDetailsAsync(
+            productDetailsParams
+        ) { billingResult, productsDetail ->
+            if (billingResult.responseCode == OK && productsDetail != null) {
+                val productDetailsBuilder = BillingFlowParams.ProductDetailsParams.newBuilder()
+                    .setProductDetails(productsDetail.first())
+                offerToken?.let { productDetailsBuilder.setOfferToken(it) }
+                val productDetailsParamsList =
+                    listOf(productDetailsBuilder.build())
+
+                queryAllSubsPurchaseHistory(ProductType.SUBS.value) { subscriptionHistory ->
+                    val oldPurchaseToken: String = getOldPurchaseToken(subscriptionHistory, oldProductId)
+
+                    val billingFlowParams =
+                        BillingFlowParams.newBuilder()
+                            .setProductDetailsParamsList(productDetailsParamsList)
+                            .setSubscriptionUpdateParams(
+                                BillingFlowParams.SubscriptionUpdateParams.newBuilder()
+                                    .setOldPurchaseToken(oldPurchaseToken)
+                                    .setReplaceProrationMode(prorationMode).build()
+                            ).build()
+
+                    billingClient?.launchBillingFlow(mContext as Activity, billingFlowParams)
+                        .takeIf { billingResult ->
+                            billingResult?.responseCode != OK
+                        }?.let { billingResult ->
+                            Log.e(TAG, "Failed to launch billing flow $billingResult")
+                            val billingError = CBException(
+                                ErrorDetail(
+                                    message = GPErrorCode.LaunchBillingFlowError.errorMsg,
+                                    httpStatusCode = billingResult.responseCode
+                                )
+                            )
+                            if (ProductType.SUBS == this.purchaseProductParams.product.type) {
+                                purchaseCallBack?.onError(
+                                    billingError
+                                )
+                            } else {
+                                oneTimePurchaseCallback?.onError(
+                                    billingError
+                                )
+                            }
+                        }
+                }
+            } else {
+                Log.e(TAG, "Failed to fetch product :" + billingResult.responseCode)
+                if (ProductType.SUBS == this.purchaseProductParams.product.type) {
+                    purchaseCallBack?.onError(throwCBException(billingResult))
+                } else {
+                    oneTimePurchaseCallback?.onError(throwCBException(billingResult))
+                }
```

Comment:
Do we need to handle for `oneTimePurchaseCallback`? Since OTP cannot be changed.

Reply:
Missed it!!... Removed now.